lib/ansible/modules/apt.py

```diff
@@ -690,28 +690,45 @@ def parse_diff(output):
     return {'prepared': '\n'.join(diff[diff_start:diff_end])}
 
 
-def mark_installed_manually(m, packages):
+def mark_installed(m, packages, manual=True):
     if not packages:
         return
 
+    if manual:
+        mark_state = "unmarkauto"
+        mark_msg = "manually"
+        mark_op = "manual"
+    elif not manual:
+        mark_state = "markauto"
+        mark_msg = "auto"
+        mark_op = "auto"
+
     apt_mark_cmd_path = m.get_bin_path("apt-mark")
 
     # https://github.com/ansible/ansible/issues/40531
     if apt_mark_cmd_path is None:
-        m.warn("Could not find apt-mark binary, not marking package(s) as manually installed.")
+        m.warn("Could not find apt-mark binary, not marking package(s) as %s installed." % mark_msg)
         return
 
-    cmd = "%s manual %s" % (apt_mark_cmd_path, ' '.join(packages))
+    cmd = "%s %s %s" % (apt_mark_cmd_path, mark_op, ' '.join(packages))
```

Comment:
This command should be built as a list, since there's no need to run it through shell parsing:

```suggestion
    cmd = [apt_mark_cmd_path, mark_op] + packages
```